lambda_src/drivers/process_https.py

```diff
@@ -59,22 +58,29 @@ def process_row(
     req_headers.setdefault('User-Agent', 'GEFF 1.0')
     req_headers.setdefault('Accept-Encoding', 'gzip')
 
+    # We look for an auth header and if found, we parse it from its encoded format
     if auth is not None:
         auth = decrypt_if_encrypted(auth)
         assert auth is not None
+
         req_auth = (
             loads(auth)
             if auth.startswith('{')
             else parse_header_dict(auth)
             if auth
             else {}
         )
+        auth_host = req_auth.get('host')
+
+        if not auth_host:
+            raise ValueError('\'auth\' is missing the \'SMTP_HOST\' key.')
 
-        if 'host' in req_auth and not req_host:
-            req_host = req_auth['host']
+        # We reject the request if the 'auth' is not pinned to a host or doesn't match the pinned host
+        if auth_host != req_host:
```

Comment:
keep `auth_host is Null` in this check?

Reply:
that's the opposite of what we want -- if the auth host is missing, we want to send an un-authenticated https request

Reply:
in general, can we keep refactoring of https in a separate PR? this one is supposed to be a change to the SMTP handler

Reply:
ACK. I was making the pinning uniform for both. I thought this PR was related to our pinning discussion hence the changes related to that.

Reply:
thanks, yes -- we want user / password to be optional in SMTP. we want to retain the optional auth in https

Reply:
It's done. Let me know if the changes look okay. Opened a new PR for the HTTP changes.

Reply:
Tested with secrets manager. 
![image](https://user-images.githubusercontent.com/72515998/137415327-8efcbb21-7fcc-407f-b7a6-82da434d78b4.png)